### PR TITLE
Add an internal class to serialize/deserialize VisitorID

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package com.adobe.marketing.mobile.internal.util;
+
+import com.adobe.marketing.mobile.VisitorID;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VisitorIDSerializer {
```
Comment (prudrabhat) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:20]: Nit: Can you please add Java doc for this public class and  methods

**Review by prudrabhat — APPROVED** [review]

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        Map<String, Object> data = new HashMap<>();
+        data.put(ID, visitorID.getId());
+        data.put(ID_ORIGIN, visitorID.getIdOrigin());
+        data.put(ID_TYPE, visitorID.getIdType());
+        data.put(STATE, visitorID.getAuthenticationState().getValue());
+        return data;
+    }
+
+    public static List<VisitorID> convertToVisitorIds(List<Map> data) {
+        List<VisitorID> visitorIDList = new ArrayList<>();
```
Comment (praveek) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:36]: Can you add null checks?

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Map;
+
+public class VisitorIDSerializer {
+    private static final String ID = "ID";
+    private static final String ID_ORIGIN = "ID_ORIGIN";
+    private static final String ID_TYPE = "ID_TYPE";
+    private static final String STATE = "STATE";
+
+    public static Map<String, Object> convertVisitorId(VisitorID visitorID) {
+        Map<String, Object> data = new HashMap<>();
```
Comment (praveek) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:27]: Add null checks or mark with `NonNull` annotation

Comment (praveek) [reply code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:27]: nit: Why are both these methods different (Serializing VisitorID but deserializing list of VisitorIDs) ?

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java (hunk 2 above)
Comment (yangyansong-adbe) [reply code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:68]: Added the `@NonNull` annotation

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java (hunk 3 above)
Comment (yangyansong-adbe) [reply code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:38]: Added a new method to serialize `VisitorID` list.

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java [hunk 4]
```diff
@@ -32,7 +43,28 @@ public static Map<String, Object> convertVisitorId(VisitorID visitorID) {
...
+    /**
+     * Serializes the given {@link VisitorID} list to a {@link Map} list.
+     *
+     * @param visitorIDList a list of {@link VisitorID} instances to serialize
+     * @return a list of {@link Map} representing the given {@link VisitorID} properties
+     */
+    public static List<Map<String, Object>> convertVisitorIds(@NonNull List<VisitorID> visitorIDList) {
+        List<Map<String, Object>> data = new ArrayList<>();
+        for (VisitorID vId : visitorIDList) {
+            data.add(VisitorIDSerializer.convertVisitorId(vId));
```
Comment (praveek) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:55]: Can you add a null check to be safe here?

File: code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java [hunk 5]
```diff
@@ -32,7 +43,28 @@ public static Map<String, Object> convertVisitorId(VisitorID visitorID) {
...
+    /**
+     * Deserializes the given map {@link List} to a list of {@link VisitorID}.
+     *
+     * @param data a list of {@link Map} to deserialize
+     * @return a list of {@link VisitorID} that was deserialized from the property {@link Map}
+     */
+    public static List<VisitorID> convertToVisitorIds(@NonNull List<Map> data) {
         List<VisitorID> visitorIDList = new ArrayList<>();
         for (Map item : data) {
             String id = String.valueOf(item.get(ID));
```
Comment (praveek) [inline code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/util/VisitorIDSerializer.java:70]: Can you add a null check to be safe here?

**Review by praveek — APPROVED** [review]

**Pull request merged**